flink-runtime/src/main/java/org/apache/flink/runtime/metrics/groups/TaskManagerJobMetricGroup.java

```diff
@@ -61,9 +58,7 @@ public TaskManagerJobMetricGroup(
 			JobID jobId,
 			@Nullable String jobName) {
 
-		super(registry, jobId, jobName, scopeFormat.formatScope(parent, jobId, jobName));
-
-		this.parent = checkNotNull(parent);
+		super(registry, checkNotNull(parent), jobId, jobName, scopeFormat.formatScope(parent, jobId, jobName));
```

Comment:
`parent` is also used in `scopeFormat.formatScope`, as such i wanted the check before that call is made.